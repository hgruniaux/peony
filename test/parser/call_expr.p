//# RUN: peony %s

fn foo() {}
fn bar(a: i32) {}

fn test() {
    foo();
    bar(5);

    foo(5);
    //~^ ERROR: too many arguments to function 'fn foo() -> void'
    bar();
    //~^ ERROR: too few arguments to function 'fn bar(i32) -> void'

    let x = 5;
    x();
    //~^ ERROR: 'x' cannot be used as a function
    5();
    //~^ ERROR: expression cannot be used as a function

    bar(true);
    //~^ ERROR: mismatched types; expected 'i32', found 'bool'
}
