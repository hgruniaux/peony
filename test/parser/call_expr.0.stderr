<source>:10:8: error: too many arguments to function 'fn foo() -> void'
   10 |     foo(5);
      |        ^
<source>:12:8: error: too few arguments to function 'fn bar(i32) -> void'
   12 |     bar();
      |        ^
<source>:16:6: error: 'x' cannot be used as a function
   16 |     x();
      |     ~^
<source>:18:6: error: expression cannot be used as a function
   18 |     5();
      |     ~^
<source>:21:9: error: mismatched types; expected 'i32', found 'bool'
   21 |     bar(true);
      |         ~~~~
